lib/pbench/server/auth/exceptions.py

```diff
@@ -0,0 +1,21 @@
+from http import HTTPStatus
+
+
+class KeycloakError(Exception):
+    def __init__(self, http_status: int, message: str = None):
+        self.http_status = http_status
+        self.message = message if message else HTTPStatus(http_status).phrase
+
+    def __repr__(self) -> str:
+        return f"Keycloak error {self.http_status} : {str(self)}"
+
+    def __str__(self) -> str:
+        return self.message
+
+
+class KeycloakConfigError(KeycloakError):
+    pass
+
+
+class KeycloakConnectionError(KeycloakError):
+    pass
```

Comment:
> you've declared two subclasses, but only one of them is actually used.

Yeah, it was because I wrote these before their respective use cases. I just thought these 2 would be common error scenarios we might encounter. I do think with the recent refactor where I am getting all the `well-known` URIs from the dedicated oidc endpoint, we don't need the ConfigError. 

I guess at the moment we don't need any subclasses, you are right, just a generic `OIDCError` should suffice. I also changed the name from Keycloak to oidc since they not tied to our keycloak integration.